### ISSUE TITLE
NullReferenceException after upgrading to 2.2.8

### ISSUE THREAD

**Issue by twenzel** [huorswords/Microsoft.Extensions.Logging.Log4Net.AspNetCore#60]
After upgrading to the latest version 2.2.8 I get following error:

```
System.AggregateException: An error occurred while writing to logger(s). (Object reference not set to an instance of an object.) ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at IEnumerable<IDisposable> Microsoft.Extensions.Logging.Scope.Registers.Log4NetEnumerableScopedRegister.AddToScope(object state)+MoveNext()
   at new Microsoft.Extensions.Logging.Scope.Log4NetScope(object scope, Log4NetScopeRegistry registry)
   at Log4NetScope Microsoft.Extensions.Logging.Scope.Log4NetScopeFactory.BeginScope<TScope>(TScope scope)
   at IDisposable Microsoft.Extensions.Logging.Log4NetLogger.BeginScope<TState>(TState state)
   at IDisposable Microsoft.Extensions.Logging.Logger.BeginScope<TState>(TState state)
   --- End of inner exception stack trace ---
   at IDisposable Microsoft.Extensions.Logging.Logger.BeginScope<TState>(TState state)
   at Context Microsoft.AspNetCore.Hosting.Internal.HostingApplication.CreateContext(IFeatureCollection contextFeatures)
   at async Task Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests<TContext>(IHttpApplication<TContext> application)
   at async Task Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync<TContext>(IHttpApplication<TContext> application)
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at IEnumerable<IDisposable> Microsoft.Extensions.Logging.Scope.Registers.Log4NetEnumerableScopedRegister.AddToScope(object state)+MoveNext()
   at new Microsoft.Extensions.Logging.Scope.Log4NetScope(object scope, Log4NetScopeRegistry registry)
   at Log4NetScope Microsoft.Extensions.Logging.Scope.Log4NetScopeFactory.BeginScope<TScope>(TScope scope)
   at IDisposable Microsoft.Extensions.Logging.Log4NetLogger.BeginScope<TState>(TState state)
   at IDisposable Microsoft.Extensions.Logging.Logger.BeginScope<TState>(TState state)<---
```

**Comment by samuelcadieux** [comment]
Same over here, switching back to 2.2.6 solved the issue.

**Comment by huorswords** [comment]
Hello @twenzel & @samuelcadieux ,

Thank you by opening the issue.

It will be very helpful if you could you be more specific regarding the exception message with...
- Target framework
- Scenario details

I'm trying to reproduce it on the sample application available on this repository, but unfortunately for sure I'm not reproducing the same situation that you are.

Thank you in advance by your help.

**Comment by huorswords** [comment]
Note: I suspect that your scenario is an async/await one, but will be very helpful if you can give me a minimum reproduce scenario to work with.

**Comment by ozanmut** [comment]
same error here. I debugged and pinpointed the error, sharing it as screenshot (error occurs in Log4NetEnumerableScopedRegister.cs keyValuePair.Key="CorrelationId", keyValuePair.Value=null, this null causes this problem):
![image](https://user-images.githubusercontent.com/14005027/50741316-c2e1db80-120c-11e9-9226-e0bf45ec548f.png)

startup looks like this:
![image](https://user-images.githubusercontent.com/14005027/50741338-fb81b500-120c-11e9-8e77-6752d6b20a8a.png)

**Comment by huorswords** [comment]
Thank you very much @ozanmut.

I will protect this scenario in order to avoid perform the ToString() when the object is null.

**Comment by huorswords** [comment]
This quick fix has been released on version 2.2.9. Available now at nuget.org. Can you test it and let me know the results?

Thank you all by your help.

**Comment by w-bonelli** [comment]
2.2.9 resolves this for my case. Thanks @huorswords!
